components/tidb_query/src/codec/mysql/json/json_depth.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use super::Json;
+
+impl Json {
+    pub fn depth(&self) -> i64 {
+        match &self {
+            Json::Object(_) | Json::Array(_) => depth_json(self),
+            _ => 1,
+        }
+    }
+}
+
+pub fn depth_json(j: &Json) -> i64 {
+    let mut subdepth = 0;
+    let result = match *j {
+        Json::Array(ref array) => {
+            let mut result = 0;
+            for child in array {
+                let child_depth = depth_json(child);
+                if result < child_depth {
+                    result = child_depth;
+                }
+            }
+            result
+        }
+
+        Json::Object(ref map) => {
+            let mut result = 0;
+            for key in map.keys() {
+                let value_depth = depth_json(&map[key]);
```

Comment:
Why don't  just travers on values... I think it's faster because it avoid unnecessary querying `BTreeMap`.

Reply:
Done